packages/components/src/components/NFT/NFT.tsx

```diff
@@ -15,24 +22,66 @@ export interface NFTProps {
  * TODO: Add a component that fetchs the NFT data from the blockchain and uses this component to
  * display it
  */
-export const NFT = ({
-  tokenId,
-  name,
-  imageUrl,
-  assetContractName,
-  assetContractSymbol,
-}: NFTProps) => {
-  const displayName = name || tokenId;
+export const NFT = ({ contractAddress, tokenId }: NFTProps) => {
+  const _isMounted = useRef(true);
+  const [nftData, setNftData] = React.useState<NFTData>();
+
+  const fetchNFTData = useCallback(async () => {
+    const response = await (
+      await fetch(`https://api.opensea.io/api/v1/asset/${contractAddress}/${tokenId}/`)
```

Comment:
don't forget error handling!

Also, we should decide as a group if we wanna use `.then` or `async/awaits` for fetching. i'm fine either way, but let's be consistent

Reply:
Yeah, will add error handling.

I prefer `async/await`. Wbu? And @crondinini?

Reply:
me too, let's go with `async/await`. i was probably a bit lazy on my other PR to not factor that out. Maybe it's worth moving all the api requests into their own file too, that way any component can import them.

Reply:
I'm more of an async/await kinda of person. But I'm fine either way as long as we have consistency.

In this case though, I don't think we need the 2 awaits. As far as I know `.json()` does not return a promise. I also think we don't need `useCallback` at this point. 

Could do:
```js
const response = (
      await fetch(`https://api.opensea.io/api/v1/asset/${contractAddress}/${tokenId}/`)
    ).json();
```
or 

```js
  const fetchNFTData = async () => {
    const fetchResponse = await fetch(`https://api.opensea.io/api/v1/asset/${contractAddress}/${tokenId}/`);

    if (_isMounted.current) {
      const response = fetchResponse.json();
      setNftData({
        tokenId: response.token_id,
        imageUrl: response.image_url,
        name: response.name,
        assetContractName: response.asset_contract.name,
        assetContractSymbol: response.asset_contract.symbol,
        animationUrl: response.animation_url,
      });
    }
  };
```

Reply:
you should need 2 awaits. `json()` is async, because parsing the response can be expensive for extremely large responses